etc/update-deps.pl

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use 5.010;
+
+use YAML::PP;
+use Data::Dumper;
+use Mojo::File qw(path);
+use FindBin qw($Bin $Script);
+
+my $scriptname = __FILE__;
+my $yamlfile   = "dependencies.yaml";
+my $file       = "$Bin/../$yamlfile";
+my $specfile   = "$Bin/../openQA.spec";
+my $cpanfile   = "$Bin/../cpanfile";
+
+my $data = YAML::PP->new->load_file($file);
+my $spec = path($specfile)->slurp;
+
+my @spectargets = qw(assetpack build client common devel main worker test);
+my @cpantargets = qw(assetpack client common main test worker);
+#@targets = qw(client);
+
+for my $target (@spectargets) {
+    say "Target $target";
+    my $deps     = $data->{$target . '_requires'};
+    my $prefix   = "%define ${target}_requires";
+    my $specline = $prefix;
+    for my $key (sort keys %$deps) {
+        my $version = $deps->{$key};
+        if (ref $version) {
+            $version = $version->{rpm};
+        }
+        $specline .= " $key";
+        if ($deps->{$key}) {
+            $specline .= " $version";
+        }
+    }
+    my $comment = "# The following line is generated from $yamlfile";
+    if ($spec =~ s/^# .*generated.*\n^$prefix.*/$comment\n$specline/m) {
+        next;
+    }
+    # No comment above the line yet
+    unless ($spec =~ s/^$prefix.*/$comment\n$specline/m) {
+        warn "/^$prefix/ not found in $specfile";
```

Comment:
This should probably fail? Considering the list of targets is explicit

Reply:
right. fixed!